app/models/concerns/course/assessment/questions_concern.rb

```diff
@@ -20,4 +20,24 @@ def attempt(submission)
       attempting_answers.fetch(question) { question.attempt(submission) }
     end
   end
+
+  # Return the question at the given step.
+  #
+  # @param [Course::Assessment::Submission] submission The submission which contains the answers.
+  # @return [Array<Course::Assessment::Question>] The question at the given the step. The latest
+  #   unfinished question will be returned if the question at the step is not accessible.
+  def step(submission, current_step)
+    current_step = current_step.to_i
+
+    while current_step > 1
+      current_step -= 1
+      if first && first.finished_by?(submission)
```

Comment:
Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`.